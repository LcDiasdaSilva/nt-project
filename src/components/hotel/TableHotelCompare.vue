<template>
  <div>Tabela de comparação</div>
</template>
<script setup lang="ts">
import type { HotelsInterface} from '../../'
  import { defineProps } from "vue";
  const props = defineProps<{
    hotels: HotelsInterface[];
  }>();
</script>
