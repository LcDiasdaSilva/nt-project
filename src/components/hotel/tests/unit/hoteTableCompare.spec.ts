import { mount } from "@vue/test-utils";
import TableHotelCompare from "../../HotelTableCompare.vue";


jest.mock("vue-progressive-image", () => ({
  ProgressiveImage: {
    template: '<img :src="src" />',
  },
}));

describe("TableHotelCompare", () => {
  it("matches the snapshot", () => {
   
    const hotels = [
      {
        id: 1,
        name: "Hotel A",
        image: "hotel-a",
        description: "Description of Hotel A",
        address: { street: "Street A", city: "City A" },
        price: 100,
      },
      {
        id: 2,
        name: "Hotel B",
        image: "hotel-b",
        description: "Description of Hotel B",
        address: { street: "Street B", city: "City B" },
        price: 200,
      },
    ];

    const wrapper = mount(TableHotelCompare, {
      props: {
        hotels,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
