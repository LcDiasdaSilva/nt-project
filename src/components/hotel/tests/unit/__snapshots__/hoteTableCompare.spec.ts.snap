// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHotelCompare matches the snapshot 1`] = `
<div class="table-compare">
  <table class="table-auto table">
    <thead>
      <tr></tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <div><img src="/images/hotel-a.png" object-cover=""></div>
        </td>
        <td>
          <div><img src="/images/hotel-b.png" object-cover=""></div>
        </td>
      </tr>
      <tr>
        <td>
          <div class="table-compare__title">Hotel A</div>
        </td>
        <td>
          <div class="table-compare__title">Hotel B</div>
        </td>
      </tr>
      <tr>
        <td>
          <div class="table-compare__description">
            <div>Description of Hotel A</div>
          </div>
        </td>
        <td>
          <div class="table-compare__description">
            <div>Description of Hotel B</div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div>Street A</div>
          <div>City A</div>
        </td>
        <td>
          <div>Street B</div>
          <div>City B</div>
        </td>
      </tr>
      <tr>
        <td>
          <div class="table-compare__price"> Diária: R$&nbsp;100,00</div>
        </td>
        <td>
          <div class="table-compare__price"> Diária: R$&nbsp;200,00</div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
